service/api/client/client.go

```diff
@@ -158,6 +210,7 @@ func (c *Client) Subscribe(ctx context.Context, opts ...core.SubOption) (<-chan
 	if err != nil {
 		return nil, err
 	}
+	threads := make(map[thread.ID]*symmetric.Key) // Follow-key cache
```

Comment:
As records come in, we need to fetch the follow key for the thread.... so, keeping a cache of those keys here. See below...